evalml/objectives/standard_metrics.py

```diff
@@ -901,6 +902,40 @@ def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
         )
 
 
+class SilhouetteCoefficient(UnsupervisedLearningObjective):
+    """Silhouette Coefficient for clustering. Does not require ground truth values."""
+
+    name = "Silhouette Coefficient"
+    greater_is_better = True
+    score_needs_proba = False
+    perfect_score = 1.0
+    is_bounded_like_percentage = False
+    expected_range = [-1, 1]
+    requires_y_true = False
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
```

Comment:
Is it possible to make `y_true` default to None here, or are we going to expect users to pass in `None` as the `y_true` value whenever using this?

Reply:
The problem is that we can't change the default value of `y_true` without breaking the api signature we inherit from ObjectiveBase. If `UnSupervisedLearningObjective`s never needed `y_true` maybe we can modify the implementation of `score` in `UnsupervisedLearningObjective` to always pass in None for `y_true` but I think right now we have to force users to pass in `y_true=None` .

Reply:
Yeah, I was on the fence about this. It does feel a little unwieldy to ask users to pass in `None` as the `y_true` every time, but I wanted to keep the `objective_function` signature intact. Really though, it does make more sense to set defaults in this case. I'll change this.